.editorconfig

```diff
@@ -24,3 +24,39 @@ indent_size = 2
 # JSON files
 [*.json]
 indent_size = 2
+
+# Dotnet code style settings:
+# Avoid this. unless absolutely necessary
+# Use language keywords instead of BCL types for type references and method calls
+# Prefer to generate C#6 and C#7 features but do not require
+[*.{cs, vb}]
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+dotnet_style_qualification_for_method = false:suggestion
+dotnet_style_qualification_for_event = false:suggestion
+dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
+dotnet_style_predefined_type_for_member_access = true:suggestion
+dotnet_style_object_initializer = true:suggestion
+dotnet_style_collection_initializer = true:suggestion
+dotnet_style_coalesce_expression = true:suggestion
+dotnet_style_null_propagation = true:suggestion
+
+# CSharp code style settings:
+# Only use var when it's obvious what the variable type is
+# Prefer to generate C#6 and C#7 features but do not require
+[*.cs]
+csharp_style_var_for_locals = true:none
+csharp_style_var_for_built_in_types = true
```

Comment:
@CyrusNajmabadi i was applying the "compiler rules" at the repo level and then have editorconfigs for the IDE projects overwrite these settings